graphql_api/types/commit/commit.py

```diff
@@ -190,12 +189,31 @@ def resolve_path_contents(commit: Commit, info, path: str = None, filters=None):
                 f"missing coverage for report with components: {component_filter}"
             )
 
-        commit_report = components.component_filtered_report(
-            commit_report, filtered_components
-        )
+        component_flags = []
+        for component in filtered_components:
+            component_paths.extend(component.paths)
+            component_flags.extend(
+                component.get_matching_flags(commit_report.flags.keys())
+            )
+
+    if flags_filter:
+        flags_filter = set(flags_filter)
+        if not (flags_filter & set(commit_report.flags)):
+            return UnknownFlags()
+        if not (flags_filter & set(component_flags)):
+            return UnknownFlags(
+                f"unknown flags for report with components: {component_filter}"
+            )
+        flags_filter = list(flags_filter & set(component_flags))
+    else:
+        flags_filter = list(set(flags_filter) | set(component_flags))
```

Comment:
if there's not flags_filter then we know it's def gonna be `component_flags` filter so no need for the OR here